### PR TITLE
Fully qualify container images

### LINKED ISSUE #3423: Container resources (e.g. redis) not compatible with Podman
I noticed that you can run [Aspire on Podman](https://learn.microsoft.com/en-us/dotnet/aspire/fundamentals/setup-tooling?tabs=dotnet-cli%2Cunix#container-runtime), so I tried the [sample application](https://learn.microsoft.com/en-us/dotnet/aspire/get-started/build-your-first-aspire-app) with the environment variable set. 

When running the application you get the following errors logged
```
fail: Aspire.Hosting.Dcp.dcpctrl.ContainerReconciler[0]
      could not create the container    {"Container": {"name":"cache"}, "Reconciliation": 2, "error": "CreateContainer command returned error code 125 Stdout: '' Stderr: 'Error: short-name resolution enforced but cannot prompt without a TTY\n'"}
fail: Aspire.Hosting.Dcp.dcpctrl.ContainerReconciler[0]
      container has failed to start     {"Container": {"name":"cache"}, "Reconciliation": 4, "error": "CreateContainer command returned error code 125 Stdout: '' Stderr: 'Error: short-name resolution enforced but cannot prompt without a TTY\n'"}
^C%
```
And shows on the Dashboard
image

The issue is the image names need an explicit prefix for Podman, unlike Docker wich assumes `docker.io` prefix if none exists. So for full Podman support, all image names from docker hub will need to be explicit with `docker.io/redis` as the image name.

Verified by changing the sample Program code to:

```c#
// var cache = builder.AddRedis("cache");

var cache = builder.AddResource(new RedisResource("cache"))
                    .WithAnnotation(new EndpointAnnotation(System.Net.Sockets.ProtocolType.Tcp, port: null, containerPort: 6379))
                    .WithAnnotation(new ContainerImageAnnotation { Image = "docker.io/redis", Tag = "7.2.4" })
                    .PublishAsContainer();
```
Which then works perfectly:

image

Environment:
Fedora 39
dotnet SDK 8.0.100
Aspire version: 8.0.0-preview.4.24156.9+692dc41a65da572a7df25d53a9f2880afe59fdd
Podman v4.9.4

### PR DESCRIPTION
**Pull request by eerhardt**
Podman doesn't default to the docker.io registry when no registry is listed. Fully qualify the container image and registry to avoid any issues with assuming a reference.

Fix #3423
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/3500)

### CODE REVIEW

File: src/Aspire.Hosting.Kafka/KafkaBuilderExtensions.cs [hunk 1]
```diff
@@ -26,6 +26,7 @@ public static IResourceBuilder<KafkaServerResource> AddKafka(this IDistributedAp
         return builder.AddResource(kafka)
             .WithEndpoint(targetPort: KafkaBrokerPort, port: port, name: KafkaServerResource.PrimaryEndpointName)
             .WithImage("confluentinc/confluent-local", "7.6.0")
+            .WithImageRegistry("docker.io")
```
Comment (mitchdenny) [inline src/Aspire.Hosting.Kafka/KafkaBuilderExtensions.cs:29]: Capture this in a class like the other ones?

Comment (eerhardt) [reply src/Aspire.Hosting.Kafka/KafkaBuilderExtensions.cs]: Done.

**Review by karolz-ms — APPROVED** [review]

**Review by davidfowl — APPROVED** [review]

**Pull request merged**